aws/rust-runtime/aws-auth/src/provider/lazy_caching.rs

```diff
@@ -3,53 +3,52 @@
  * SPDX-License-Identifier: Apache-2.0.
  */
 
+//! Lazy, caching, credentials provider implementation
+
 use crate::provider::cache::Cache;
 use crate::provider::time::TimeSource;
-use crate::provider::{AsyncProvideCredentials, BoxFuture, CredentialsResult};
+use crate::provider::timeout::Timeout;
+use crate::provider::{
+    AsyncProvideCredentials, AsyncSleep, BoxFuture, CredentialsError, CredentialsResult,
+};
 use std::sync::Arc;
 use std::time::Duration;
 use tracing::{trace_span, Instrument};
 
-const DEFAULT_REFRESH_TIMEOUT: Duration = Duration::from_secs(5);
+const DEFAULT_LOAD_TIMEOUT: Duration = Duration::from_secs(5);
 const DEFAULT_CREDENTIAL_EXPIRATION: Duration = Duration::from_secs(15 * 60);
 const DEFAULT_BUFFER_TIME: Duration = Duration::from_secs(10);
 
-// TODO: Implement async runtime-agnostic timeouts
-// TODO: Add catch_unwind() to handle panics
-// TODO: Update doc comment below once catch_unwind and timeouts are implemented
-// TODO: Update warning not to use this in the STS example once it's prod ready
-
 /// `LazyCachingCredentialsProvider` implements [`AsyncProvideCredentials`] by caching
 /// credentials that it loads by calling a user-provided [`AsyncProvideCredentials`] implementation.
 ///
 /// For example, you can provide an [`AsyncProvideCredentials`] implementation that calls
 /// AWS STS's AssumeRole operation to get temporary credentials, and `LazyCachingCredentialsProvider`
 /// will cache those credentials until they expire.
-///
-/// # Note
-///
-/// This is __NOT__ production ready yet. Timeouts and panic safety have not been implemented yet.
 pub struct LazyCachingCredentialsProvider {
     time: Box<dyn TimeSource>,
+    sleeper: Box<dyn AsyncSleep>,
```

Comment:
Not a hard/fast rule, but based on my observations, usually w/ Rust, `*er` tends to be avoided as a suffix.